pkg/scheduler/reconcile_on_location.go

```diff
@@ -19,33 +19,239 @@ package scheduler
 import (
 	"context"
 
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/klog/v2"
 
 	kcpcache "github.com/kcp-dev/apimachinery/v2/pkg/cache"
+	schedulingv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/scheduling/v1alpha1"
+	workloadv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/workload/v1alpha1"
+	"github.com/kcp-dev/logicalcluster/v3"
+
+	edgev1alpha1 "github.com/kcp-dev/edge-mc/pkg/apis/edge/v1alpha1"
 )
 
 func (c *controller) reconcileOnLocation(ctx context.Context, locKey string) error {
 	logger := klog.FromContext(ctx)
-	ws, _, name, err := kcpcache.SplitMetaClusterNamespaceKey(locKey)
+	lws, _, lName, err := kcpcache.SplitMetaClusterNamespaceKey(locKey)
 	if err != nil {
 		logger.Error(err, "invalid Location key")
 		return err
 	}
-	logger = logger.WithValues("workspace", ws, "location", name)
+	logger = logger.WithValues("locationWorkspace", lws, "location", lName)
 	logger.V(2).Info("reconciling")
 
 	/*
 		On location change:
-		- find all its st(s)
+		1) find all ep(s) that selected loc
+
+		2a) find all its st(s), and update store --- how to get its previous st(s)?
+		2b) find all ep(s) that selecting loc, and update store
 
-		- find all its ep(s), and update store
-		- for each of its obsolete ep, remove all sp(s)
-		- for each of its ongoing ep, update all sp(s)
-		- for each of its new ep, add all sp(s)
+		3a) for each of its obsolete ep, remove all sp(s)
+		3b) for each of its ongoing ep, update all sp(s)
+		3c) for each of its new ep, add all sp(s)
 
 		Need data structure:
 		- map from a location to its eps, say 'epsBySelectedLoc'
+
+		TODO(waltforme): Maybe I can merge 3b) and 3c)
 	*/
 
+	loc, err := c.locationLister.Cluster(lws).Get(lName)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return nil
```

Comment:
There is work to be done in this case, right?  Updating the SinglePlacementSlice objects that formerly referenced the Location that is now deleted.

Reply:
Yes, completed this in PR #267 .